Add ability for generate-config script to add directly to ESR

matteyu opened the pull request:
-Changed python interpreter from 2.7 to 3.6
-Added web3 functionality
-Changed logging to info
-Web3/Infura account keys read off external files

matteyu committed:
Add ability for generate-config script to add directly to ESR

amastracci commented on peerlist-server/generate-config.py:
```diff
@@ -1,17 +1,389 @@
 
+# required to have these files in the peerlist-server directory
+# infura = infura link including API key
+i = open("infura", "r")
```
Do these need to be global?

matteyu replied:
Yep.  They're required by the acct variable and contract_address variable which is used by multiple functions in the script.  It represents the ropsten address and ropsten contract address.

amastracci commented on peerlist-server/generate-config.py:
```diff
@@ -40,6 +412,37 @@ def get_node_uuid(node_id, working_directory):
+    ####Add node to BluzelleDockerSwarm
+    nonce = w3.eth.getTransactionCount(acct.address)
+    node_name = "node_{}".format(node_id)
+    node_host = ([l for l in ([ip for ip in socket.gethostbyname_ex(socket.gethostname())[2] 
```
Move this to function

amastracci replied:
Give some thought to failure cases and where we see console errors within AWS

matteyu replied:
Moved.  Changes reflected with the new commit.

For viewing errors on console, it's a configuration through the cloudformation script to include CloudWatch.

Commit:
Adding changes for signalfx integration

matteyu committed:
Docker file changes

matteyu committed:
-Merge feature/amastracci/KEP-1319 branch
-switched host ip retrieval to a function
-modified dockerfile and docker-compose in swarm-node to include NUM-NODES variable

matteyu committed:
Ignore peerlist-server folder and top docker-compose file

matteyu committed:
Store peerlist server in archive folder (ignored)
